app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -141,17 +147,21 @@ public Subpopulation getSubpopulation(StudyIdentifier studyId, SubpopulationGuid
     }
     
     @Override
-    public void deleteSubpopulation(StudyIdentifier studyId, SubpopulationGuid subpopGuid) {
+    public void deleteSubpopulation(StudyIdentifier studyId, SubpopulationGuid subpopGuid, boolean physicalDelete) {
         Subpopulation subpop = getSubpopulation(studyId, subpopGuid);
-        if (subpop == null || subpop.isDeleted()) {
+        if (subpop == null || (!physicalDelete && subpop.isDeleted())) {
             throw new EntityNotFoundException(Subpopulation.class);
         }
```

Comment:
Unit tests for:
* logically deleting a logically deleted subpop
* physically deleting a logically deleted subpop